vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml

```diff
@@ -1,299 +1,249 @@
+        inputs:
+          SourceFolder: '$(Build.SourcesDirectory)/iot-e2e-tests/android/app/build/outputs/apk'
+          Contents: |
+            *.*
+          TargetFolder: '$(Build.ArtifactStagingDirectory)'
+        continueOnError: true
+        condition: always()
+
+      - task: PublishPipelineArtifact@0
+        inputs:
+          artifactName: 'androidBuildFiles'
+          targetPath: 'iot-e2e-tests/android/app/build/outputs/apk'
+
+  - job: AndroidTest
+    timeoutInMinutes: 80
+    pool:
+      vmImage: 'macOS-latest'
+    strategy:
+      maxParallel: 17
+      matrix:
+        TestGroup1:
+          ANDROID_TEST_GROUP_ID: TestGroup1
+        TestGroup2:
+          ANDROID_TEST_GROUP_ID: TestGroup2
+        TestGroup3:
+          ANDROID_TEST_GROUP_ID: TestGroup3
+        TestGroup4:
+          ANDROID_TEST_GROUP_ID: TestGroup4
+        TestGroup5:
+          ANDROID_TEST_GROUP_ID: TestGroup5
+        TestGroup6:
+          ANDROID_TEST_GROUP_ID: TestGroup6
+        TestGroup7:
+          ANDROID_TEST_GROUP_ID: TestGroup7
+        TestGroup8:
+          ANDROID_TEST_GROUP_ID: TestGroup8
+        TestGroup9:
+          ANDROID_TEST_GROUP_ID: TestGroup9
+        TestGroup10:
+          ANDROID_TEST_GROUP_ID: TestGroup10
+        TestGroup11:
+          ANDROID_TEST_GROUP_ID: TestGroup11
+        TestGroup12:
+          ANDROID_TEST_GROUP_ID: TestGroup12
+        TestGroup13:
+          ANDROID_TEST_GROUP_ID: TestGroup13
+        TestGroup14:
+          ANDROID_TEST_GROUP_ID: TestGroup14
+        TestGroup15:
+          ANDROID_TEST_GROUP_ID: TestGroup15
+        TestGroup16:
+          ANDROID_TEST_GROUP_ID: TestGroup16
+        TestGroup17:
+          ANDROID_TEST_GROUP_ID: TestGroup17
+
+    displayName: Android Test
+    dependsOn: AndroidBuild
+    steps:
+      - task: PowerShell@2
+        displayName: 'determine if testing needed'
+        condition: always()
+        inputs:
+          targetType: 'filePath'
+          filePath: ./vsts/determine_if_android_test_group_needs_to_run.ps1
+        env:
+          TEST_GROUP_ID: $(ANDROID_TEST_GROUP_ID)
+          IS_BASIC_TIER_HUB: $(IS-BASIC-TIER-HUB)
+          TARGET_BRANCH: $(System.PullRequest.TargetBranch)
+
+      - task: DownloadPipelineArtifact@0
+        condition: eq(variables['task.android.needToRunTestGroup'], 'yes')
+        inputs:
+          artifactName: 'androidBuildFiles'
+          targetPath: $(Build.SourcesDirectory)/iot-e2e-tests/android/app/build/outputs/apk
+
+      - task: Bash@3
+        displayName: 'Start Android Emulator'
+        inputs:
+          targetType: 'filePath'
+          filePath: '$(Build.SourcesDirectory)/vsts/StartEmulator.sh'
+
+      - task: Bash@3
+        displayName: 'Run tests on emulator'
+        inputs:
+          targetType: 'filePath'
+          filePath: '$(Build.SourcesDirectory)/vsts/RunTestsOnEmulator.sh'
+        env:
+          TEST_GROUP_ID: $(ANDROID_TEST_GROUP_ID)
```

Comment:
Note that the android environment variable values are set during build time, not runtime. The APKs that are generated from a build contain the environment variables such as iothub connection string. As such, there is no need to include them here, too